.github/workflows/prepare_release.yml

```diff
@@ -0,0 +1,72 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: false
+        default: ''
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+      base-branch:
+        description: 'The branch that will be used as the origin for the release branch.'
+        required: false
+        default: ''
+        type: string
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+  BUNDLE_WITH: release
+  GIT_AUTHOR_NAME: Release Automation
+  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
+  BLACKSMITH_FULL_VERSION: "${{ inputs.version }}"
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Prepare Release'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
```

Comment:
Perforce makes a full clone with:

```yaml
      - name: "Checkout"
        uses: "actions/checkout@v4"
        with:
          fetch-depth: 0
```

In my tests that wasn't required. Maybe it's for some edge cases. Based on their git history I didn't find any explanation.